app/controllers/cinema_halls_controller.rb

```diff
@@ -4,17 +4,20 @@ class CinemaHallsController < ApplicationController
   before_action :authenticate_user!
 
   def index
+    authorize CinemaHall
     cinema_halls = CinemaHall.all
     render json: cinema_halls
   end
 
   def show
     cinema_hall = CinemaHall.find(params[:id])
+    authorize cinema_hall
     render json: cinema_hall
   end
 
   def create
     cinema_hall = CinemaHall.new(cinema_hall_params)
+    authorize cinema_hall
```

Comment:
Like I noticed on call, you can also just put here:
```ruby
def create
  authorize CinemaHall
```
cause this means also
```ruby
def create
  authorize CinemaHall, :create? # name of action to authorize is automatically assumed from name of controller method
```

I guess that it's much cleaner way of solving this.